src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export const fetchUserPosts = createAsyncThunk<Post[], number>(
+  'posts/fetchUserPosts',
+  userId => {
+    const response = getUserPosts(userId);
```

Comment:
The `getUserPosts` function is being called synchronously. Since it's likely an asynchronous operation fetching data from an API, it should be awaited to ensure the response is received before returning it.